Add BLEnder config (#6609)
mozilla/blurts-server: .blender/fix-dependabot-prompt.md

# BLEnder: Fix CI failures on Dependabot PRs

You are fixing CI failures on a Dependabot pull request in the Firefox Monitor (Blurts Server) repository.

## Context

{{PR_TITLE}}

Failing checks:
{{FAILING_CHECKS}}

## CI logs

{{CI_LOGS}}

## Your task

Fix the CI failures caused by this dependency update. Make the minimum change needed. Do not refactor unrelated code.

## Common fix patterns for this repo

### Linters and formatters

**ESLint** (JavaScript/TypeScript):
```
npm run build-glean
npm run build-nimbus
npx eslint --fix <file>
# Or for all files:

> npx eslint .

Reviewer: ```suggestion
npx eslint --max-warnings=0 .
```

be aligned with what ci actually runs

```
CI runs: `npm run lint` (which includes eslint, prettier, stylelint, tsc, typegen, Nimbus validation)

**Prettier** (formatting):
```
npx prettier --write <file>
# Or check only:

> npx prettier --check .

Reviewer: ```suggestion
npx prettier --check ./src
```

This might actually create false positives

```

**Stylelint** (CSS/SCSS):
```
npx stylelint --allow-empty-input --fix "**/*.{scss,css}"
```

**TypeScript type checking**:
```

> npx tsc --noEmit

Reviewer: ```suggestion
npx next typegen && npx tsc -p tsconfig.json --noEmit
```

lets specify the json file with -p. We have multiple tsconfig

```

### Running the full lint suite (as CI does)

```bash
npm ci
npm run build-glean
npm run build-nimbus

> npm run lint

Reviewer: ```suggestion
npm run lint
node src/scripts/build/checkNodeVersionAlignment.js
node src/scripts/build/checkGithubActionsBestPractices.js
```

```

> `npm run lint` runs in sequence: stylelint, prettier --check, eslint, tsc --noEmit, typegen validation, Nimbus validation, and a Node version alignment check.

Reviewer: ```suggestion
CI runs `npm run lint`, then separately runs the Node version alignment and GitHub Actions best-practices checks.
```

This is factually wrong

### Build check

CI also runs a full build:
```bash
npm ci
npm run build-glean
npm run build
```
`npm run build` runs: Glean build, Nimbus build, Next.js build, cronjob build.

### Node version alignment check

Part of `npm run lint`. The script `src/scripts/build/checkNodeVersionAlignment.js` verifies that the Node version is consistent across `package.json` (volta + engines), `Dockerfile`, `netlify.toml`, esbuild target, and all GitHub Actions workflows. If you update `.github/workflows/` files or `package.json`, ensure the node version stays at `20.20.x`.

## Strategy

1. If you know which check failed, run that check first to reproduce the error.
2. If unclear, run the relevant checks in this order:
   - `npm run build-glean && npm run build-nimbus && npm run lint`
   - `npm run build-glean && npm test`
3. Read the error output. Identify the root cause.
4. Make the fix. Run the check again to confirm.
5. If you cannot fix it, say so. Do not guess.
6. You have a limited number of turns. Be direct. Do not explore the codebase beyond what is needed to fix the specific error.

## Rules

- Only change files related to the dependency update failure.
- Do not add new dependencies.

> - Do not modify CI configuration files.

Reviewer: ```suggestion
-  Do not modify CI configuration files unless the Dependabot update or failing check is specifically for CI, GitHub Actions, Docker, or Node-version alignment.
```

### Linters and formatters

**ESLint** (JavaScript/TypeScript):

```
npm run build-glean
npm run build-nimbus
npx eslint --fix <file>
# Or for all files:
npx eslint --max-warnings=0 .
```

**Prettier** (formatting):

```
npx prettier --write <file>
# Or check only:
npx prettier --check ./src
```

**Stylelint** (CSS/SCSS):

```
npx stylelint --allow-empty-input --fix "**/*.{scss,css}"
```

**TypeScript type checking**:

```
npx next typegen && npx tsc -p tsconfig.json --noEmit
```

Note: The repo has multiple tsconfig files (`tsconfig.json` and `tsconfig.cronjobs.json`). CI uses `-p tsconfig.json`.

### Running the full lint suite (as CI does)

```bash
npm ci
npm run build-glean
npm run build-nimbus
npm run lint
node src/scripts/build/checkNodeVersionAlignment.js
node src/scripts/build/checkGithubActionsBestPractices.js
```

`npm run lint` runs in sequence: `stylelint '**/*.scss'`, `prettier --check './src'`, `eslint --max-warnings=0 .`, `next typegen`, `tsc -p tsconfig.json --noEmit`, and `validate-nimbus`. The two `node` scripts above run as separate CI steps after `npm run lint`.

### Unit tests

```bash
npm ci
npm run build-glean
npm test
```

> CI runs `npm test` which calls `npm run build-glean && vitest run`.

Reviewer: ```suggestion
CI runs `npm run build-glean` before `npm test`; `npm test` calls `npm run build-nimbus && vitest run`.
```
